src/code/main.c

```diff
@@ -58,7 +58,7 @@ void Main(void* arg) {
     SystemHeap_Init(sysHeap, gSystemHeapSize); // initializes the system heap
     if (osMemSize >= 0x800000) {
         debugHeap = SysCfb_GetFbEnd();
-        debugHeapSize = (s32)(0x80600000 - debugHeap);
+        debugHeapSize = (s32)(K0BASE + 0x600000 - debugHeap);
```

Comment:
This cast doesn't seem to be required (apart from this just being a generally weird calculation to do).